spec/vmfloaty/ssh_spec.rb

```diff
@@ -4,46 +4,109 @@
       return {
         os_types.keys[0] => { 'hostname' => ['abs-hostname.delivery.puppetlabs.net'] },
         'ok' => true
       }
-    end
 
-    {
-      os_types.keys[0] => { 'hostname' => 'vmpooler-hostname' },
-      'domain' => 'delivery.puppetlabs.net',
-      'ok' => true
-    }
+    elsif os_types.keys[0] == 'vmpooler_api_v2_host_string'
+      return {
+        os_types.keys[0] => { 'hostname' => ['vmpooler-v2-hostname.delivery.puppetlabs.net'] },
+        'ok' => true
+      }
+
+    else
+      return {
+        os_types.keys[0] => { 'hostname' => 'vmpooler-v1-hostname' },
+        'domain' => 'delivery.puppetlabs.net',
+        'ok' => true
+      }
+    end
   end
 
   def type
     return 'abs' if os_types == 'abs_host_string'
-    return 'vmpooler' if os_types == 'vmpooler_host_string'
+    return 'vmpooler' if os_types == 'vmpooler_api_v1_host_string' || os_types == 'vmpooler_api_v2_host_string'
+  end
+
+  def wait_for_request(verbose, requestid)
+    return true
   end
 end
 
 describe Ssh do
   before :each do
   end
 
-  it 'gets a hostname string for abs' do
-    verbose = false
-    service = ServiceStub.new
-    host_os = 'abs_host_string'
-    use_token = false
-    cmd = Ssh.command_string(verbose, service, host_os, use_token)
-    expect(cmd).to match(/ssh root@abs-hostname.delivery.puppetlabs.net/)
+  context "for pooled requests" do
+    it 'gets a hostname string for abs' do
+      verbose = false
+      service = ServiceStub.new
+      host_os = 'abs_host_string'
+      use_token = false
+      cmd = Ssh.command_string(verbose, service, host_os, use_token)
+      expect(cmd).to match(/ssh root@abs-hostname.delivery.puppetlabs.net/)
+    end
+
+    it 'gets a hostname string for vmpooler api v1' do
+      verbose = true
+      service = ServiceStub.new
+      host_os = 'vmpooler_api_v1_host_string'
+      use_token = false
+      cmd = Ssh.command_string(verbose, service, host_os, use_token)
+      expect(cmd).to match(/ssh root@vmpooler-v1-hostname.delivery.puppetlabs.net/)
+    end
+
+    it 'gets a hostname string for vmpooler api v2' do
+      verbose = false
+      service = ServiceStub.new
+      host_os = 'vmpooler_api_v2_host_string'
+      use_token = false
+      cmd = Ssh.command_string(verbose, service, host_os, use_token)
+      expect(cmd).to match(/ssh root@vmpooler-v2-hostname.delivery.puppetlabs.net/)
+    end
   end
 
-  it 'gets a hostname string for vmpooler' do
-    verbose = false
-    service = ServiceStub.new
-    host_os = 'vmpooler_host_string'
-    use_token = false
-    cmd = Ssh.command_string(verbose, service, host_os, use_token)
-    expect(cmd).to match(/ssh root@vmpooler-hostname.delivery.puppetlabs.net/)
+  context "for ondemend requests" do
+    let(:service) { ServiceStub.new }
+    let(:url) { 'http://pooler.example.com' }
+
+    it 'gets a hostname string for abs' do
+      verbose = false
+      host_os = 'abs_host_string'
+      use_token = false
+      ondemand = true
```

Comment:
I think for ABS the ondemand flag should be false since ABS is the level of abstraction and itself checks if the pool requested is pooled or ondemand.

Reply:
I guess its fine to test in case the ondemand flag is sent to an abs service? In practice this flag should be only for direct vmpooler access, and originally we did not provide it intentionally so that our users would have to switch to ABS.

Reply:
its fine then, since we dont enforce that.